test/functional/ui/decorations_spec.lua

```diff
@@ -866,6 +866,36 @@ end]]
   end)
 end)
 
+describe('extmark conceal', function()
+  local screen, ns
+  before_each( function()
+    clear()
+    screen = Screen.new(50, 5)
+    screen:attach()
+    screen:set_default_attr_ids {
+      [0] = {bold=true, foreground=Screen.colors.Blue};
+      [1] = {background=Screen.colors.DarkGrey, foreground=Screen.colors.LightGrey};
+    }
+
+    ns = meths.create_namespace 'test'
+  end)
+
+  it('conceals', function()
+    insert([[
+foo
+]])
```

Comment:
for future reference: insert()..rawfeed() does `dedent()` so the string literal can be nicely indented. Also insert() takes a list of lines.